src/handler.ts

```diff
@@ -146,38 +152,29 @@ const handleStream = async (
 					}
 				})
 
-				if (init.value !== undefined && init.value !== null) {
-					if (
-						typeof init.value === "object"
+				if (init?.value !== undefined && init?.value !== null)
+					controller.enqueue(
+						Buffer.from(
+							`event: message\ndata: ${JSON.stringify(init.value)}\n\n`
```

Comment:
Would make sense to check typeof object before using JSON.stringify

Reply:
If you don't always call JSON.stringify and the user yields a string that contains 2 new lines the response will break the text/event-stream protocol. If you always call JSON.stringify you can be sure that all the new lines are encoded with \n and put in a single line.

The parsing in Eden also becomes simpler because you know messages are always a JSON encoded string instead of using heuristics.